lib/parser/provenance.h

```diff
@@ -97,6 +104,7 @@ class OffsetToProvenanceMappings {
   void Put(ProvenanceRange);
   void Put(const OffsetToProvenanceMappings &);
   ProvenanceRange Map(std::size_t at) const;
+  std::optional<std::size_t> ReverseMap(Provenance at) const;
```

Comment:
You can omit the name `at` and we typically do so unless it has documentary value.